inference-engine/thirdparty/clDNN/src/layout_optimizer.cpp

```diff
@@ -827,8 +911,35 @@ layout layout_optimizer::get_expected_layout(layout const& current_layout,
     auto expected_tensor = current_layout.size;
     auto expected_data_type = current_layout.data_type;
     auto expected_format = current_layout.format;
-
-    if (_optimization_attributes.b_fs_zyx_fsv16_network &&
+    auto input_layout = node.get_dependency(0).get_output_layout();
+    auto is_2d = input_layout.format.spatial_num() == 2;
+    bool use_onednn_impls = _optimization_attributes.use_onednn_impls;
+
+    bool onednn_valid_dt = input_layout.data_type == data_types::i8 ||
+                           input_layout.data_type == data_types::u8 ||
+                           input_layout.data_type == data_types::f16;
+
+    bool onednn_valid_params = onednn_valid_dt &&
+                               input_layout.size.feature[0] >= 16 &&
+                               prim->groups == 1 &&
+                               /* get_post_ops_count(node) <= 10 */ false &&
```

Comment:
I think this should be either `get_post_ops_count() <= 32` or nothing, in other places as well..